### PR TITLE
sasview: init at 4.1.2

### PR DESCRIPTION
**Pull request by rprospero**
###### Motivation for this change

[Sasview](http://www.sasview.org) is a data fitting and analysis package for small angle scattering data.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by FRidh — CHANGES_REQUESTED** [review]
Libraries that are in `python-packages.nix` should be used. If those versions are not compatible, then they can be overridden.

**Comment by rprospero** [comment]
I'm now using html5lib from python-packages and merely overriding the version.  This has also allowed me to move xhtml2pdf into the python-packages.nix, instead of making it part of sasview.

I've also done just a bit of cleanup as I was manually importing licenses into a couple of the packages for no good reason.

File: pkgs/applications/science/misc/sasview/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{lib, fetchgit, gcc, python}:
+
+python.pkgs.buildPythonApplication rec {
+  name = "sasview-${version}";
```
Comment (FRidh) [inline pkgs/applications/science/misc/sasview/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed. Instead use `pname` for just the name.

File: pkgs/applications/science/misc/sasview/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    xhtml2pdf];
+
+  src = fetchgit {
+    url = "https://github.com/SasView/sasview.git";
+    rev = "v${version}";
+    sha256 ="05la54wwzzlkhmj8vkr0bvzagyib6z6mgwqbddzjs5y1wd48vpcx";
+  };
+
+  patches = [./pyparsing-fix.patch ./local_config.patch];
+  sandbox = true;
```
Comment (FRidh) [inline pkgs/applications/science/misc/sasview/default.nix]: sandbox?

File: pkgs/development/python-modules/bumps/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, buildPythonPackage, fetchPypi, six}:
+
+buildPythonPackage rec {
+  name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/bumps/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

File: pkgs/development/python-modules/sasmodels/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{lib, fetchgit, buildPythonPackage, pytest, numpy, scipy, matplotlib, docutils}:
+
+buildPythonPackage rec {
+  name = "sasmodels-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/sasmodels/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

File: pkgs/development/python-modules/sasmodels/default.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  version = "0.96";
+
+  propagatedBuildInputs = [docutils matplotlib numpy pytest scipy];
```
Comment (FRidh) [inline pkgs/development/python-modules/sasmodels/default.nix]: test dependencies should not be propagated

File: pkgs/development/python-modules/xhtml2pdf/default.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
+{lib, fetchPypi, buildPythonPackage, html5lib, httplib2, nose, pillow, pypdf2, reportlab}:
+
+let
+  #xhtml2pdf specifically requires version "1.0b10" of html5lib
+  html5 = html5lib.overrideAttrs( oldAttrs: rec{
```
Comment (FRidh) [inline pkgs/development/python-modules/xhtml2pdf/default.nix]: We permit only a single version within `python-packages.nix`, so no overriding here.

File: pkgs/top-level/python-packages.nix [hunk 7]
```diff
@@ -12276,6 +12278,26 @@ in {
     };
   };
 
+  periodictable = buildPythonPackage rec{
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: We are moving Python expressions out of `pkgs/top-level/python-packages.nix` into `pkgs/development/python-modules/<module>/default.nix`. 

Please move the expression there, and call it from `pkgs/top-level/python-packages.nix` using `callPackage ../development/python-modules/<package> { };`.

File: pkgs/top-level/python-packages.nix [hunk 8]
```diff
@@ -12276,6 +12278,26 @@ in {
...
+    name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

File: pkgs/top-level/python-packages.nix [hunk 9]
```diff
@@ -17329,6 +17353,29 @@ in {
     };
   };
 
+  unittest-xml-reporting = buildPythonPackage rec {
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: We are moving Python expressions out of `pkgs/top-level/python-packages.nix` into `pkgs/development/python-modules/<module>/default.nix`. 

Please move the expression there, and call it from `pkgs/top-level/python-packages.nix` using `callPackage ../development/python-modules/<package> { };`.

File: pkgs/top-level/python-packages.nix [hunk 10]
```diff
@@ -17329,6 +17353,29 @@ in {
...
+    name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/xhtml2pdf/default.nix (hunk 6 above)
Comment (FRidh) [reply pkgs/development/python-modules/xhtml2pdf/default.nix]: Considering sasview is the only consumer of this package, I suggest moving the expression there.

**Comment by dotlambda** [comment]
@matthewbauer This one created a conflict. I manually merged master into staging right now.

**Pull request merged**